scripts/python/inv_reserve_ipmi_pxe_ips.py

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/env python
+# Copyright 2018 IBM Corp.
+#
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import nested_scopes, generators, division, absolute_import, \
+    with_statement, print_function, unicode_literals
+
+import xmlrpclib
+from netaddr import IPNetwork
+from pyghmi.ipmi import command as ipmi_command
+from pyghmi import exceptions as pyghmi_exception
+from time import time
+
+from lib.config import Config
+from lib.inventory import Inventory
+import lib.genesis as gen
+import lib.utilities as util
+from ipmi_power_off import ipmi_power_off
+from lib.exception import UserException
+import lib.logger as logger
+
+DNSMASQ_TEMPLATE = '/etc/cobbler/dnsmasq.template'
+COBBLER_USER = gen.get_cobbler_user()
+COBBLER_PASS = gen.get_cobbler_pass()
+WAIT_TIME = 1200
+POWER_TIME_OUT = gen.get_power_time_out()
+POWER_WAIT = gen.get_power_wait()
+SLEEP_TIME = gen.get_power_sleep_time()
+
+
+class IPGenerator(object):
```

Comment:
@rayjh: 'IPGenerator' is probably a poor name for this class since it's not a python generator (and doesn't return one). It keeps track of IP addresses as they're assigned to various IPMI/PXE interfaces, which could be on a common or separate subnets. If IPMI and PXE interfaces are on the same subnet I can use the 'ipmiNetwork' and 'pxeNetwork' vars to point to a single 'IPGenerator' object to ensure no duplicate IPs are assigned. For distinct IPMI and PXE subnets distinct 'IPGenerator' objects are used.

I'll change the name of this class in a subsequent commit.